be2-scala/src/main/scala/ch/epfl/pop/decentralized/Monitor.scala

```diff
@@ -82,9 +85,19 @@ final case class Monitor(
     case ConnectionMediator.Ping() =>
       log.info("Received ConnectionMediator ping")
       connectionMediatorRef = sender()
-      new Thread(new FileMonitor(connectionMediatorRef)).start()
+      fileMonitor = new FileMonitor(this.self)
+      new Thread(fileMonitor).start()
 
-    case _ => /* DO NOTHING */
+    case msg: ConnectionMediator.ConnectTo =>
+      connectionMediatorRef ! msg
+  }
+
+  override def postStop(): Unit = {
+    if (fileMonitor != null) {
+      if (!fileMonitor.running.compareAndSet(true, false)) {
```

Comment:
Both boolean evaluation could be merged into one, using "&&" operator which short circuits the evaluation if first result is false. Meaning that if fileMonitor is null, fileMonitor.running.compareAndSet will not be evaluated.

Reply:
No, because if we merge it will log an error message when fileMonitor is null, which can happen if no ConnectionMedatior.Ping() is received, which is not wanted